src/lib/parquet/src/Flow/Parquet/ParquetFile/RowGroupBuilder/ColumnChunkBuilder.php

```diff
@@ -22,31 +22,43 @@ public function addRow(mixed $data) : void
 
     /**
      * @psalm-suppress PossiblyNullArgument
+     *
+     * @return array<ColumnChunkContainer>
      */
-    public function flush(int $fileOffset) : ColumnChunkContainer
+    public function flush(int $fileOffset) : array
     {
-        $pageContainer = (new DataPagesBuilder($this->data))->build($this->column);
+        $offset = $fileOffset;
+        $columnChunkContainers = [];
+        $previousChunkData = null;
+
+        foreach (\array_chunk($this->data, 1000) as $dataChunk) {
```

Comment:
this is hardcoded for now, but the idea is to take a chunk of data, build a data page, and check if it's bigger than 8Kb (if not, drop it, merge data from the next page, and try over again). 
Of course size of data page and that chunk size will come from the configuration with default values:
- data page size = 8Kb
- data page probe rows count = 1_000